### DISCUSSION TITLE
Possible to ignore part totally?

### DISCUSSION THREAD

**Discussion by Hipshot-zz** [QodotPlugin/Qodot#145] [General]
Like, could I make a func_ignore and have that not be added to godot upon building the map with qodot?

**Comment by Andicraft** [comment]
I’m not 100% sure but I bet you could do this with a custom entity where the tool script just deletes itself after being spawned

**Reply by Hipshot-zz** [reply]
Yea, probably. Not sure how I would do that, but it would be a nice addition anyway to have such support. Then you can leave "copy paste" ojects and test areas in the same map and enable or disable them as you need them.

Could be nice with such a setting in the properties panel, where you also set skip texture and such. "Dont import this group name" or so =)

**Comment by sinewavey** [comment]
Why not just extend QodotMap, and in `build_entity_nodes` for example, look if some property you also add (could even be through a base class, DEBUG or something, right? special flavor of appearance when you think about it) is of a desirable value.

like, it already handles this a bit, with a line like:
```
...
if entity_definition is QodotFGDPointClass:
...
  elif entity_definition.node_class != "":
      node.queue_free()
      node = ClassDB.instantiate(entity_definition.node_class)
...
```

Perhaps something like:

```
	for entity_idx in range(0, entity_dicts.size()):
		var entity_dict := entity_dicts[entity_idx] as Dictionary
		var properties := entity_dict['properties'] as Dictionary

                if 'debug_skip_generation' in properties:
                 # negate here, etc

```

Many options to hard exclude things I think, beyond just freeing them in ready.

**Comment by RhapsodyInGeek** [comment]
Entity definitions have a property called `transient_node`. If this property is set to true, the generated node will be removed at the end of Qodot map building.

https://github.com/QodotPlugin/Qodot/blob/33503596f2624ce3b2dec1c16f70f3e7b6d9ab4f/addons/qodot/src/resources/game-definitions/fgd/qodot_fgd_class.gd#L27

https://github.com/QodotPlugin/Qodot/blob/33503596f2624ce3b2dec1c16f70f3e7b6d9ab4f/addons/qodot/src/nodes/qodot_map.gd#L1246
